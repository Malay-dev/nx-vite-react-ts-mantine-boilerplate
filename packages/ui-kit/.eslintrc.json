{
  "extends": ["../../.eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts", "*.tsx"],
      "rules": {},
      "parserOptions": {
        "project": ["tsconfig.*?.json"]
      }
    }
  ]
}
