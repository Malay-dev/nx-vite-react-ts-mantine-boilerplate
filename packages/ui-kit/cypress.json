{
  "fileServerFolder": ".",
  "fixturesFolder": "./cypress/fixtures",
  "integrationFolder": "./cypress/integration",
  "modifyObstructiveCode": false,
  "supportFile": "./cypress/support/index.ts",
  "pluginsFile": "./cypress/plugins/index.ts",
  "video": false,
  "videosFolder": "./dist/cypress/ui-kit/videos",
  "screenshotsFolder": "./dist/cypress/ui-kit/screenshots",
  "chromeWebSecurity": false,
  "env": {
    "ALLOW_SCREENSHOT": true,
    "updateSnapshots": true
  }
}
