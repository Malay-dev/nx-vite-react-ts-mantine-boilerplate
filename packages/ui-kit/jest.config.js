module.exports = {
  displayName: 'ui-kit',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'ts-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/packages/ui-kit',
};
