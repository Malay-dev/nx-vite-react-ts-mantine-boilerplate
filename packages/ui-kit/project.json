{
  "name": "ui-kit",
  "sourceRoot": "packages/ui-kit/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "type-check": {
      "executor": "nx:run-commands",
      "options": {
        "cwd": "./packages/ui-kit",
        "commands": [
          {
            "command": "tsc --project tsconfig.lib.json --noEmit"
          }
        ]
      }
    },
    "e2e": {
      "executor": "nx:run-commands",
      "options": {
        "cwd": "./packages/ui-kit",
        "commands": [
          {
            "command": "cypress run --component --config-file=cypress.config.ts"
          }
        ]
      }
    },
    "build": {
      "executor": "nx:run-commands",
      "options": {
        "cwd": "./packages/ui-kit",
        "commands": [
          {
            "command": "vite build --emptyOutDir"
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/ui-kit/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/ui-kit/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "nx:run-commands",
      "options": {
        "cwd": "./packages/ui-kit",
        "commands": [
          {
            "command": "vitest run --config ./vitest.config.ts --coverage --passWithNoTests"
          }
        ]
      }
    },
    "storybook": {
      "executor": "@nrwl/storybook:storybook",
      "options": {
        "uiFramework": "@storybook/react",
        "port": 4400,
        "configDir": "packages/ui-kit/.storybook"
      },
      "configurations": {
        "ci": {
          "quiet": true
        }
      }
    },
    "storybook:build": {
      "executor": "@nrwl/storybook:build",
      "outputs": ["{options.outputDir}"],
      "options": {
        "uiFramework": "@storybook/react",
        "configDir": "packages/ui-kit/.storybook",
        "outputDir": "packages/ui-kit/dist/storybook"
      },
      "configurations": {
        "ci": {
          "quiet": true
        }
      }
    },
    "storybook:serve": {
      "executor": "nx:run-commands",
      "options": {
        "cwd": "./packages/ui-kit",
        "commands": [
          {
            "command": "npx http-server dist/storybook"
          }
        ]
      }
    }
  }
}
