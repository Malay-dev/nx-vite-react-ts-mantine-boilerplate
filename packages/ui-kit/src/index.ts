export { Button } from './lib/Button'
export { Error } from './lib/Error'
export { Input } from './lib/Input'
export { Loader } from './lib/Loader'
export { Pagination } from './lib/Pagination'
export { Select } from './lib/Select'
export { Space } from './lib/Space'
export { Table } from './lib/Table'
export type { SelectItem } from '@mantine/core'
