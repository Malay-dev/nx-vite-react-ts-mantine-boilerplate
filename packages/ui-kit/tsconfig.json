{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "jsx": "react",
    "allowJs": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "forceConsistentCasingInFileNames": true,
    "strict": true,
    "noImplicitOverride": true,
    "noPropertyAccessFromIndexSignature": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true
  },
  "files": [
    "../../node_modules/@nrwl/react/typings/image.d.ts",
    "../../node_modules/cypress/types/cypress-global-vars.d.ts"
  ],
  "references": [
    {
      "path": "./tsconfig.lib.json"
    },
    {
      "path": "./tsconfig.spec.json"
    },
    {
      "path": "./.storybook/tsconfig.json"
    }
  ]
}
