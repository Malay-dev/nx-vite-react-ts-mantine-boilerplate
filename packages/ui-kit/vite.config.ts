import { defineConfig } from 'vite'

import path from 'node:path'
import dts from 'vite-dts'
import viteReact from '@vitejs/plugin-react'

const isExternal = (id: string) => !id.startsWith('.') && !path.isAbsolute(id)

export default defineConfig({
  esbuild: {
    jsxInject: "import React from 'react'",
  },
  build: {
    lib: {
      entry: path.resolve(__dirname, 'src/index.ts'),
      formats: ['es'],
    },
    rollupOptions: {
      external: isExternal,
    },
  },
  plugins: [
    dts(),
    viteReact({
      babel: {
        plugins: [
          [
            'babel-plugin-styled-components',
            {
              pure: true,
              ssr: true,
              displayName: true, // TODO: Need to add production settings
              fileName: true, // TODO: Need to add production settings
            },
          ],
        ],
      },
    }),
  ],
})
