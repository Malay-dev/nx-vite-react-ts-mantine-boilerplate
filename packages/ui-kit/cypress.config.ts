import { defineConfig } from 'cypress'

import viteConfig from './vite.config'

export default defineConfig({
  component: {
    specPattern: 'src/**/*.cy.{js,jsx,ts,tsx}',
    devServer: {
      framework: 'react',
      bundler: 'vite',
      viteConfig,
    },
  },
})
