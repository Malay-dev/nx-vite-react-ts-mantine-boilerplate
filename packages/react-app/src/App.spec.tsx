import React from 'react'
import { render } from '@testing-library/react'
import { describe, expect, it } from 'vitest'

import { UiKit } from '@nx-ts-vite-react-graphql-styled-monorepo-example/ui-kit'

describe('UiKit', () => {
  it('should render successfully', () => {
    const { baseElement, container } = render(<UiKit />)

    expect(baseElement).toBeTruthy()
    expect(container).toMatchSnapshot()
  })
})
