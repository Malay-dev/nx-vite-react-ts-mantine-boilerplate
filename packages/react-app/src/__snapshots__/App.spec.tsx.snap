// Vitest Snapshot v1

exports[`UiKit > should render successfully 1`] = `
<div>
  <div
    class="sc-bdvvtL gAzFMf"
    style="color: pink;"
  >
    <h1>
      Welcome to UiKit!
    </h1>
  </div>
</div>
`;
